ui/raidboss/data/03-hw/dungeon/xelphatol.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+// Xelphatol
+[{
+  zoneId: ZoneId.Xelphatol,
+  timelineFile: 'xelphatol.txt',
+  timelineTriggers: [
+    {
+      id: 'Xelphatol Short Burst',
+      regex: /Short Burst/,
+      beforeSeconds: 4,
+      condition: Conditions.caresAboutMagical(),
+      suppressSeconds: 5, // Timelines jump sometimes, so let's not be noisy.
+      response: Responses.tankBuster(),
+    },
+    {
+      // This might be better handled by collecting Swiftfeather,
+      // but a timeline trigger is easy and consistent.
+      id: 'Xelphatol On Low',
+      regex: /On Low/,
+      beforeSeconds: 5,
+      response: Responses.tankCleave(),
+    },
+    {
+      id: 'Xelphatol Ixali Aero Buster',
+      regex: /Ixali Aero \(buster\)/,
+      beforeSeconds: 4,
+      condition: Conditions.caresAboutMagical(),
```

Comment:
I think this is probably fine for a dungeon scenario, but one bad part about caresAboutMagical is that it doesn't include when the target is you.  I'm not sure how to more elegantly handle this, though.

This is maybe another case where it'd be nice to have `ConditionFuncs.caresAboutMagical(data) || ConditionFuncs.targetIsYou(data, matches)` or something.

(cc @panicstevenson FYI, just because you've run into this some lately too)

Reply:
"Fortunately" this doesn't matter for a timeline trigger, but yes, it would be nice if `ConditionFuncs` were a thing.

Reply:
Oh right, it's a timeline trigger :+1:.

Reply:
RE: `ConditionFuncs`, another alternative (potentially a stopgap) would be to add `Conditions.physicalTankbuster()` and `Conditions.magicalTankbuster()` which would be `matches.target || Conditions.caresAboutX()`. It would be better to have composable parts to use as building blocks, but I don't know how many things would benefit from this aside from Tankbusters.

Reply:
I'll think it over a little more and then open an issue about it later tonight. There's definitely something there!